src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -394,6 +451,29 @@ export default function TransactionFeedV2() {
     [data?.transactions]
   )
 
+  useEffect(
+    function trackCrossChainSwaps() {
+      if (!data?.transactions) return
+
+      const completedCrossChainSwaps = data.transactions
+        .filter(
+          (tx) =>
+            tx.status === TransactionStatus.Complete &&
+            tx.type === TokenTransactionTypeV2.CrossChainSwapTransaction
+        )
+        .map((tx) => tx.transactionHash)
+
+      const newlyCompletedCrossChainSwaps = standByTransactions.pending.filter(
+        (tx): tx is TokenExchange => completedCrossChainSwaps.includes(tx.transactionHash)
+      )
+
+      if (newlyCompletedCrossChainSwaps.length) {
+        trackCompletionOfCrossChainSwaps(newlyCompletedCrossChainSwaps)
+      }
+    },
+    [data?.transactions, standByTransactions.pending]
```

Comment:
Is there a risk that the tracking could be triggered twice if one of these deps change without being in sync with the other?

Reply:
@jeanregisser You're right! Initially, I thought is shouldn't be a problem based on how the flow of `data?.transactions` and `standByTransactions` is tightly connected. But now as I think about it, it makes a lot more sense to utilize and add an extra field to `useNewlyCompletedTransactions` as it should be more precise in signaling the change.